pkg/kubecost/allocation.go

```diff
@@ -2273,15 +2273,14 @@ func (asr *AllocationSetRange) AccumulateBy(resolution time.Duration) (*Allocati
 
 	asr.Lock()
 	defer asr.Unlock()
-
 	for i, as := range asr.allocations {
 		allocSet, err = allocSet.accumulate(as)
 		if err != nil {
 			return nil, err
 		}
 
 		if allocSet != nil {
-			currAccumulatedSum += allocSet.Window.Duration()
+			currAccumulatedSum = allocSet.Window.Duration()
```

Comment:
This doesn't seem right. Why shouldn't the addition happen?

Reply:
For example, let's say I have the following AllocationSetRange:
```
asr = [
  as1 // window 1d
  as2 // window 1d
  as3 // window 1d
]
```

I try to aggregate by `2d`. I'm pretty sure that because this line is continually resetting `currAccumulatedSum`, the resulting ASR will be:

```
[
  as // window 3d
]
```

Because the `currAccumulatedSum >= resolution` check will never be true, so we end up in the final case of `i == len(asr.allocations)-1` and accumulate all three allocation sets.

I could be wrong! I'm not sure if this situation is addressed in your unit tests.

Reply:
That was initially how I thought it worked too, and hence the bug, but when accumulating the starting window doesn't get changed and remains the same until reset by `allocSet = NewAllocationSet(time.Time{}, time.Time{})`

After each accumulation the duration (`currAccumulatedSum`) will be the window.start of the first set passed and the window.end will be the end of the last set passed, so resetting is ok here. I tested this on gke and it looks to be working correctly now.

The newest unit test was the only test to fail and catch the bug, it wasn't caught previously due to not mocking enough allocation sets (4+)

Reply:
I simplified the code further since `currAccumulateSum` is no longer needed with that bug fix and is now more readable

Reply:
Very nice catch, I like the removal of `currAccumulatedSum`.